server/embedded/src/org/labkey/embedded/ResourceType.java

```diff
@@ -0,0 +1,113 @@
+            {
+                @Override
+                protected Map<String, String> mergeWithDefaults(Map<String, String> props)
+                {
+                    Map<String, String> result = super.mergeWithDefaults(props);
+                    result.putIfAbsent("type", DataSource.class.getName());
+                    result.putIfAbsent("maxTotal", MAX_TOTAL_CONNECTIONS_DEFAULT);
+                    result.putIfAbsent("maxIdle", MAX_IDLE_DEFAULT);
+                    result.putIfAbsent("maxWaitMillis", MAX_WAIT_MILLIS_DEFAULT);
+                    result.putIfAbsent("accessToUnderlyingConnectionAllowed", ACCESS_TO_CONNECTION_ALLOWED_DEFAULT);
+                    result.putIfAbsent("validationQuery", VALIDATION_QUERY_DEFAULT);
+                    return result;
+                }
+
+                @Override
+                public void addResource(String name, Map<String, String> props, StandardContext context) throws ConfigException
+                {
+                    String displayName = props.remove("displayName");
+                    if (displayName != null)
+                    {
+                        context.addParameter(name + ":DisplayName", displayName);
+                    }
+                    String logQueries = props.remove("logQueries");
+                    if (logQueries != null)
+                    {
+                        context.addParameter(name + ":LogQueries", displayName);
+                    }
+
+                    super.addResource(name, props, context);
+                }
+            },
+    jms
+            {
+                @Override
+                protected Map<String, String> mergeWithDefaults(Map<String, String> props)
+                {
+                    Map<String, String> result = super.mergeWithDefaults(props);
+                    result.putIfAbsent("type", "org.apache.activemq.ActiveMQConnectionFactory");
+                    result.putIfAbsent("factory", "org.apache.activemq.jndi.JNDIReferenceFactory");
+                    result.putIfAbsent("description", "JMS Connection Factory");
+                    result.putIfAbsent("brokerName", "LocalActiveMQBroker");
+                    return result;
+                }
+            },
+    ldap
+            {
+                @Override
+                protected Map<String, String> mergeWithDefaults(Map<String, String> props)
+                {
+                    Map<String, String> result = super.mergeWithDefaults(props);
+                    result.putIfAbsent("type", "org.labkey.premium.ldap.LdapConnectionConfigFactory");
+                    result.putIfAbsent("factory", "org.labkey.premium.ldap.LdapConnectionConfigFactory");
+                    return result;
+                }
+            },
+    generic;
+
+    protected Map<String, String> mergeWithDefaults(Map<String, String> props)
+    {
+        Map<String, String> result = new CaseInsensitiveKeyMap<>();
+        result.putAll(props);
+        return result;
```

Comment:
`mergeWithDefaults` should handle the default `auth` property, rather than my method below.
```suggestion
        result.putAll(props);
        result.putIfAbsent("auth", "Container");
        return result;
```